fv3core/stencils/fvtp2d.py

```diff
@@ -96,58 +97,44 @@ def __call__(
         ra_y,
         fx,
         fy,
-        kstart=0,
-        nk=None,
         nord=None,
         damp_c=None,
         mass=None,
         mfx=None,
         mfy=None,
     ):
         grid = self.grid
-        if nk is None:
-            nk = grid.npz - kstart
-        kslice = slice(kstart, kstart + nk)
-        compute_origin = (grid.is_, grid.js, kstart)
         corners.copy_corners_y_stencil(
-            q, origin=(grid.isd, grid.jsd, kstart), domain=(grid.nid, grid.njd, nk)
-        )
-        self.yppm_object_in(
-            q, cry, self._tmp_fy2, grid.isd, grid.ied, kstart=kstart, nk=nk
+            q, origin=grid.full_origin(), domain=grid.domain_shape_full(add=(0, 0, 1))
         )
+        self.yppm_object_in(q, cry, self._tmp_fy2, grid.isd, grid.ied)
         self.stencil_q_i(
             q,
             grid.area,
             yfx,
             self._tmp_fy2,
             ra_y,
             self._tmp_q_i,
-            origin=(grid.isd, grid.js, kstart),
-            domain=(grid.nid, grid.njc + 1, nk),
-        )
-        self.xppm_object_ou(
-            self._tmp_q_i, crx, fx, grid.js, grid.je, kstart=kstart, nk=nk
+            origin=(grid.isd, grid.js, 0),
+            domain=(grid.nid, grid.njc + 1, grid.npz + 1),
```

Comment:
can't we also go to a `domain_shape_full(add` version of this?

Reply:
with these awkward shapes I'm not sure it's more clear, but yes for sure, can change to that